stan/math/rev/core/var.hpp

```diff
@@ -306,6 +306,24 @@ class var_value {
     }
     return os << v.val();
   }
+
+  /**
+   * Returns number of rows. Only available if `T` is a matrix.
+   * @return number of rows.
+   */
+  template <typename U = T>
```

Comment:
Do you need a require here?

Reply:
Not really, but I can add one for clarity.